lib/rules/no-insecure-url.js

```diff
@@ -18,11 +18,24 @@ const DEFAULT_EXCEPTIONS = [     // TODO: add more typical false positives such
     /^http:(\/\/|\\u002f\\u002f)schemas\.microsoft\.com(\/\/|\\u002f\\u002f)?.*/i,
     /^http:(\/\/|\\u002f\\u002f)schemas\.openxmlformats\.org(\/\/|\\u002f\\u002f)?.*/i,
     /^http:(\/|\\u002f){2}localhost(:|\/|\\u002f)*/i,
-    /^http:(\/|\\u002f){2}localhost(:|\/|\\u002f)*/i,
     /^http:(\/\/)www\.w3\.org\/1999\/xhtml/i,
     /^http:(\/\/)www\.w3\.org\/2000\/svg/i
 ];
 
+function shouldFix(context, node) {
+  let sourceCode = context.getSourceCode();
+  const text = node.parent
+    ? sourceCode.getText(node.parent)
+    : sourceCode.getText(node);

```

Comment:
I suppose you're looking at the parent in case there are comments that say "insecure"?
Please add a comment above `const text ..` to explain.

Reply:
Sure I will add a comment. It is focus on the variables name. If it is something named like 
```javascript
var insecureURL = "http://...";
```
We shouldn't upgrade it

Reply:
Ah, variable names. Great point! Pls add a comment :)